packages/test-base/src/androidAndroidTest/kotlin/io/realm/kotlin/test/shared/RealmAnyTests.kt

```diff
@@ -60,54 +69,320 @@ class RealmAnyTests {
+    }
+
+    @Test
+    fun unmanaged_float() {
+        val realmAny = RealmAny.create(42F)
+        assertEquals(42F, realmAny.asFloat())
+        assertEquals(RealmAny.create(42F), realmAny)
+        assertEquals(RealmAny.Type.FLOAT, realmAny.type)
+    }
+
+    @Test
+    fun unmanaged_double() {
+        val realmAny = RealmAny.create(42.0)
+        assertEquals(42.0, realmAny.asDouble())
+        assertEquals(RealmAny.create(42.0), realmAny)
+        assertEquals(RealmAny.Type.DOUBLE, realmAny.type)
+    }
+
+    @Test
+    fun unmanaged_realmObjectId() {
+        val objectId = ObjectId.from("000000000000000000000000")
+        val realmAny = RealmAny.create(objectId)
+        assertEquals(objectId, realmAny.asRealmObjectId())
+        assertEquals(RealmAny.create(objectId), realmAny)
+        assertEquals(RealmAny.Type.OBJECT_ID, realmAny.type)
+    }
+
+    @Test
+    fun unmanaged_objectId() {
+        val objectId = BsonObjectId("000000000000000000000000")
+        val realmAny = RealmAny.create(objectId)
+        assertEquals(objectId, realmAny.asObjectId())
+        assertEquals(RealmAny.create(objectId), realmAny)
+        assertEquals(RealmAny.Type.OBJECT_ID, realmAny.type)
+    }
+
+    @Test
+    fun unmanaged_byteArray() {
+        val byteArray = byteArrayOf(42, 41, 40)
+        val realmAny = RealmAny.create(byteArray)
+        assertContentEquals(byteArray, realmAny.asByteArray())
+        assertEquals(RealmAny.create(byteArray), realmAny)
+        assertEquals(RealmAny.Type.BYTE_ARRAY, realmAny.type)
+    }
+
+    @Test
+    fun unmanaged_realmInstant() {
+        val instant = RealmInstant.now()
+        val realmAny = RealmAny.create(instant)
+        assertEquals(instant, realmAny.asRealmInstant())
+        assertEquals(RealmAny.create(instant), realmAny)
+        assertEquals(RealmAny.Type.REALM_INSTANT, realmAny.type)
+    }
+
+    @Test
+    fun unmanaged_realmUuid() {
+        val uuid = RealmUUID.from("ffffffff-ffff-ffff-ffff-ffffffffffff")
+        val realmAny = RealmAny.create(uuid)
+        assertEquals(uuid, realmAny.asRealmUUID())
+        assertEquals(RealmAny.create(uuid), realmAny)
+        assertEquals(RealmAny.Type.REALM_UUID, realmAny.type)
+    }
+
+    @Test
+    fun unmanaged_realmObject() {
+        val obj = TestParent()
+        val realmAny = RealmAny.create(obj)
+        assertEquals(obj, realmAny.asRealmObject<TestParent>())
+        assertEquals(RealmAny.create(obj), realmAny)
+        assertEquals(RealmAny.Type.REALM_OBJECT, realmAny.type)
+    }
+
+    private fun getManagedRealmAny(block: () -> RealmAny?): RealmAny? {
+        realm.writeBlocking {
+            val realmAnyObject = copyToRealm(RealmAnyContainer())
+            realmAnyObject.anyField = block()
+        }
+
+        val container = realm.query<RealmAnyContainer>()
+            .first()
+            .find { assertNotNull(it) }
+
+        return container.anyField
+    }
+
+    @Test
+    fun managed_null() {
```

Comment:
Same, would be nice to have a single method that verifies we test all types automatically